spacy/tokens/span.pyx

```diff
@@ -422,6 +426,48 @@ cdef class Span:
         else:
             raise ValueError(Errors.E030)
 
+    @property
+    def sents(self):
+        """Obtain the sentences that contain this span. If the given span
+        crosses sentence boundaries, return all sentences it is a part of.
+
+        RETURNS (Iterable[Span]): All sentences that the span is a part of.
+
+         DOCS: https://spacy.io/api/span#sents
+        """
+        cdef int start
+        cdef int i
+
+        if not self.doc.has_annotation("SENT_START"):
+            raise ValueError(Errors.E030)
```

Comment:
This check should come after checking the user hooks - if there is a valid hook we don't have to raise the exception.

Otherwise I think this looks good!